extensions/ql-vscode/src/vscode-tests/cli-integration/remote-queries/variant-analysis-manager.test.ts

```diff
@@ -359,409 +359,418 @@ describe("Variant Analysis Manager", () => {
-        });
+        getVariantAnalysisRepoStub = jest.spyOn(
+          ghApiClient,
+          "getVariantAnalysisRepo",
+        );
+        getVariantAnalysisRepoResultStub = jest.spyOn(
+          ghApiClient,
+          "getVariantAnalysisRepoResult",
+        );
+      });
 
-        it("should skip the download if the repository has already been downloaded", async () => {
-          // First, do a download so it is downloaded. This avoids having to mock the repo states.
-          await variantAnalysisManager.autoDownloadVariantAnalysisResult(
-            scannedRepos[0],
-            variantAnalysis,
-            cancellationTokenSource.token,
-          );
+      describe("when the artifact_url is missing", () => {
+        beforeEach(async () => {
+          const dummyRepoTask = createMockVariantAnalysisRepoTask();
+          delete dummyRepoTask.artifact_url;
 
-          getVariantAnalysisRepoStub.mockClear();
+          getVariantAnalysisRepoStub.mockResolvedValue(dummyRepoTask);
+          getVariantAnalysisRepoResultStub.mockResolvedValue(arrayBuffer);
+        });
 
+        it("should not try to download the result", async () => {
           await variantAnalysisManager.autoDownloadVariantAnalysisResult(
             scannedRepos[0],
             variantAnalysis,
             cancellationTokenSource.token,
           );
 
-          expect(getVariantAnalysisRepoStub).not.toHaveBeenCalled();
+          expect(getVariantAnalysisRepoResultStub).not.toHaveBeenCalled();
         });
+      });
 
-        it("should write the repo state when the download is successful", async () => {
-          await variantAnalysisManager.autoDownloadVariantAnalysisResult(
-            scannedRepos[0],
-            variantAnalysis,
-            cancellationTokenSource.token,
-          );
+      describe("when the artifact_url is present", () => {
+        let dummyRepoTask: VariantAnalysisRepoTask;
 
-          expect(outputJsonStub).toHaveBeenCalledWith(
-            join(
-              storagePath,
-              variantAnalysis.id.toString(),
-              "repo_states.json",
-            ),
-            {
-              [scannedRepos[0].repository.id]: {
-                repositoryId: scannedRepos[0].repository.id,
-                downloadStatus:
-                  VariantAnalysisScannedRepositoryDownloadStatus.Succeeded,
-              },
-            },
-          );
+        beforeEach(async () => {
+          dummyRepoTask = createMockVariantAnalysisRepoTask();
+
+          getVariantAnalysisRepoStub.mockResolvedValue(dummyRepoTask);
+          getVariantAnalysisRepoResultStub.mockResolvedValue(arrayBuffer);
         });
 
-        it("should not write the repo state when the download fails", async () => {
-          getVariantAnalysisRepoResultStub.mockRejectedValue(
-            new Error("Failed to download"),
-          );
+        describe("autoDownloadVariantAnalysisResult", () => {
```

Comment:
We're already in a describe block for `autoDownloadVariantAnalysisResult` as far as I can tell. Is this redundant?

Reply:
Yes, this is redundant, I must have missed this while moving the tests.